File: Ejemplos literal D/Ejemplo1.py
#Ejemplo 1
#Clase padre
class CuentadeBitcoin:
#Definimos atributos y métodos de la clase padre
    def __init__(self, saldo=0.0):
         self.saldo = saldo

    def retirar(self, cantidad):
     if self.saldo >= cantidad:
         self.saldo -= cantidad
         print("\tRetiro exitoso. Nuevo saldo:"+ str(round(self.saldo,2)))
     else: 
        print("\tNo hay suficientes fondos para retirar")
# Subclase que herada de CuentaDeBitcoin
#Definimos atributos y métodos de la clase heredada
class CuentaPremium(CuentadeBitcoin):
    def __init__(self, saldo=0.0, cashback=0.05):
        #Llamamos a la clase padre
        super().__init__(saldo)
        self.cashback = cashback
    def retirarCP(self, cantidad):
        if self.saldo >= cantidad:
            self.saldo -= cantidad
            cashback_ganado = cantidad * self.cashback
            self.saldo += cashback_ganado
            print(f"\tRetiro exitosos con cashback. Nuevo saldo:"+ str(round(self.saldo,2))+ f" (Cashback: {cashback_ganado})")
        else:
            print("\tNo hay suficientes fondos para retirar")

#Creamos instancias de los objetos
Cuenta_normal=CuentadeBitcoin(0)
Cuenta_gold=CuentaPremium(170.3)

#Imprimiendo valores
Cuenta_normal.retirar(50.0)
Cuenta_gold.retirarCP(50.0)

File: Ejemplos literal D/Ejemplo 1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ejemplo_1
{
    //Creamos la clase padre
    public class CuentadeBitcoin
        //Defomos sus propiedades y métodos
    {
        protected double Saldo;
        public CuentadeBitcoin(double SaldoInicial)
        {
            Saldo = SaldoInicial;
        }
        public void Retirar(double cantidad)
        {
            if (Saldo >= cantidad)
            {
                Saldo -= cantidad;
                Console.WriteLine($"\n\tRetiro exitoso. Nuevo saldo: {Saldo}");
            }
            else
            {
                Console.WriteLine("\tNo hay suficientes fondos para retirar");
            } 
                
        }
        //Creamos la clase heredada
        public class CuentaPremium : CuentadeBitcoin
        {
        //Definimos las propiedades y métodos
            private double porcentajecashback = 0.05;
            public CuentaPremium(double SaldoInicial) : base(SaldoInicial)
            { }
            public new void Retirar(double cantidad)
            {
                if (Saldo >= cantidad)
                {
                    //Llamamos a la clase padre
                    base.Retirar(cantidad);
                    double cashback=cantidad*porcentajecashback;
                    Saldo += cashback;
                    Console.WriteLine($"\tRetiro con cashback. Nuevo saldo {Saldo} (Cashback: {cashback})");
                }
                else
                {
                    Console.WriteLine("\tNo hay suficientes fondos para retirar");
                }
            }
        }
        public class Program
        {
            static void Main(string[] args)
            {
                //Creamos las instancias de los objetos
                CuentadeBitcoin Cuenta1 = new CuentadeBitcoin(0);
                CuentaPremium Cuenta2 = new CuentaPremium(170.3);
                //Impriendo detalles
                Cuenta1.Retirar(50.0);
                Cuenta2.Retirar(50.0);
            }
        }
    }
}
